format.php

```diff
@@ -1,248 +1,418 @@
-	}
-	
-	private function is_closed_tag()
-	{
-		$closed_tags = array(
-			'meta', 'link', 'img', 'hr', 'br', 'input',
-		);
-		
-		$current_tag = '';
-		
-		for ($input_index = $this->input_index; $input_index < strlen($this->input); $input_index++) {
-			if ($this->input[$input_index] == '<') {
-				continue;
-			} elseif (preg_match('/\s/', $this->input[$input_index])) {
-				break;
-			} else {
-				$current_tag .= $this->input[$input_index];
-			}
-		}
-		
-		if (in_array($current_tag, $closed_tags)) {
-			return true;
-		} else {
-			return false;
-		}
-	}
-	
-	private function is_inline_tag()
-	{
-		$inline_tags = array(
-			'title', 'a', 'span', 'abbr', 'acronym', 'b', 'basefont', 'bdo', 'big', 'cite', 'code', 'dfn', 'em', 'font', 'i', 'kbd', 'q', 's', 'samp', 'small', 'strike', 'strong', 'sub', 'sup', 'textarea', 'tt', 'u', 'var', 'del', 'pre',
-		);
-		
-		$current_tag = '';
-		
-		for ($input_index = $this->input_index; $input_index < strlen($this->input); $input_index++) {
-			if ($this->input[$input_index] == '<' OR $this->input[$input_index] == '/') {
-				continue;
-			} elseif (preg_match('/\s/', $this->input[$input_index]) OR $this->input[$input_index] == '>') {
-				break;
-			} else {
-				$current_tag .= $this->input[$input_index];
-			}
-		}
-		
-		if (in_array($current_tag, $inline_tags)) {
-			return true;
-		} else {
-			return false;
-		}
-	}
+    /*
+     * @var
+     */
+    private $input = null;
+
+    /*
+     * @var
+     */
+    private $output = null;
+
+    /*
+     * @var
+     */
+    private $in_tag = false;
+
+    /*
+     * @var
+     */
+    private $in_comment = false;
+
+    /*
+     * @var
+     */
+    private $in_content = false;
+
+    /*
+     * @var
+     */
+    private $inline_tag = false;
+
+    /*
+     * @var
+     */
+    private $i = 0;
+
+    /*
+     * @var
+     */
+    private $indent_depth = 0;
+
+    /*
+     * @var
+     */
+    private $indent_type = "\t";
+    
+    /**
+     * Static interface
+     * - Allows you to call the method witout initialising the class first
+     *
+     * <code>
+     *  // use spaces at 4 length
+     *  echo Format::HTML('Unformatted HTML string');
+     *
+     *  // use spaces at 2 length
+     *  echo Format::HTML('Unformatted HTML string', true, 2);
+     *
+     *  // use tabs
+     *  echo Format::HTML('Unformatted HTML string', false);
+     * </code>
+     *
+     * @param  string $input          HTML which is to be processed
+     * @param  bool   $use_spaces     Use spaces instead of tabs
+     * @param  int    $indent_length  Length of indent spacing
+     * @return string
+     */
+    public static function HTML($input, $use_spaces = true, $indent_length = 4)
+    {
+        return (new self)->process($input, $use_spaces, $indent_length);
+    }
+    
+    /**
+     * Process HTML
+     *
+     * <code>
+     *  $format = new Format();
+     *
+     *  // use spaces at 4 length
+     *  echo $format->process('Unformatted HTML string');
```

Comment:
whoops, this needs changing to `$format->html()`, same with others but doesn't affect functionality.